lib/heex_formatter/phases/format.ex

```diff
@@ -82,23 +83,42 @@ defmodule HeexFormatter.Phases.Format do
         end
       end
 
-    indentation = if self_closed?, do: state.indentation, else: state.indentation + 1
+    state =
+      if self_closed? do
+        state
+      else
+        if tag == "script" do
+          %{state | mode: :script}
+        else
+          %{state | indentation: state.indentation + 1}
+        end
+      end
+
+    %{state | html: state.html <> "\n" <> tag_opened}
+  end
 
-    %{state | html: state.html <> "\n" <> tag_opened, indentation: indentation}
+  defp token_to_string({:text, text, _meta}, %{mode: :script} = state) do
+    indent = indent_expression(state.indentation)
+    %{state | html: state.html <> indent_script(indent, text)}
   end
```

Comment:
```suggestion
  defp token_to_string(
         {:text, text, _meta},
         %{previous_token: {:tag_open, "script", _, _}} = state
       ) do
    %{state | html: state.html <> String.trim_trailing(text)}
  end
```

Then we don't need to put `mode: script` into the state 🙌🏻

Reply:
I think we cannot et rid of it because of eex in a script

Reply:
I merged like that but I applied the changes to not change anything within `script` tag. Maybe later we improve it to format as well but I'd say that is not a priority right now. 

https://github.com/feliperenan/heex_formatter/commit/2e7933ed464b2cd495e0c7f4d01e8c19e802538c